project.clj

```diff
@@ -7,7 +7,9 @@
                                    :password :env/clojars_password}}
 
   :dependencies [[org.clojure/clojure "1.12.0"]
-                 [com.stuartsierra/component "1.1.0"]]
+                 [com.stuartsierra/component "1.1.0"]
+                 [cheshire "5.13.0"]
+                 [org.clojure/data.csv "1.1.0"]]
```

Comment:
Given that CSV parsing is a whole thing, and your own library does a way better job of dealing with that I'd remove support for CSVs, EDN, plain text & JSON will cover most of everyone's needs.